spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -74,7 +74,10 @@
 	 */
 	private final String[] headersToEmbed;
 
-	private final ProvisioningProvider<C, P> provisioningProvider;
+	/**
+	 * {@link ProvisioningProvider} delegated by the downstream binder implementations.
+	 */
+	protected final ProvisioningProvider<C, P> provisioningProvider;
```

Comment:
This doesn't make me happy, because in the target `Binder` impl I have to cast to target `ProvisioningProvider` impl.
Adding one more generic argument - `PP extends ProvisioningProvider<C, P>` would make the trick.
But do we really need to worry about?

Just my 5 cents...